server/playground.rs

```diff
@@ -0,0 +1,139 @@
+use hubcaps::gists;
+use iron::{status, IronResult, Request, Response};
+use router::Router;
+use serde::{Deserialize, Serialize};
+use serde_json;
+use std::collections::HashMap;
+use std::convert::TryFrom;
+use std::io::Read;
+use tokio::runtime::Runtime;
+
+use crate::errors::Error;
+use crate::json;
+use crate::middleware::GistsMiddleware;
+
+#[derive(Debug, Serialize, Deserialize)]
+struct Playground {
+    id: Option<String>,
+    messages: String,
+    variables: serde_json::Value,
+    setup: serde_json::Value,
+}
+
+pub fn get(req: &mut Request) -> IronResult<Response> {
+    let gists_middleware = req.extensions.get::<GistsMiddleware>().unwrap();
+    let gists = &gists_middleware.gists;
+    let params = req.extensions.get::<Router>().unwrap();
+    let id = params.find("id").expect("No route parameter called id");
+    let mut rt = match Runtime::new() {
+        Ok(rt) => rt,
+        Err(_) => return json::error(status::ServiceUnavailable, Error::Runtime),
+    };
```

Comment:
```rust
let mut rt = Runtime::new().unwrap_or_else(|_| json::error(status::ServiceUnavaialble, Error::Runtime));
```

Reply:
I'd like to benefit from early returns in every step here. As soon as something fails, the server can respond with a `5xx` or `4xx` status. IIUC, `unwrap_or_else` doesn't let me benefit from early returns. It would also require that the type returned by `json::error` is the same as expected by the `let` binding; here it's a `Runtime`.

I think an alternative would be to chain a number of `add_then` called, one for each step.